sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala

```diff
@@ -121,3 +121,24 @@ object LeftSemiOrAnti {
     case _ => None
   }
 }
+
+object AsOfJoinDirection {
+
+  def apply(direction: String): AsOfJoinDirection = {
+    direction.toLowerCase(Locale.ROOT).replace("_", "") match {
```

Comment:
Why do we need to replace `_` with ""?

Reply:
I just followed what `JoinType` does, but yeah, I don't think we need it.